### PR TITLE
Update log format while checking the state of a resource

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
This PR updates the log format while checking for the resource by printing the name or ID of the resource.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Update log format while checking the state of a resource
```

### CODE REVIEW

File: cloud/scope/powervs_cluster.go [hunk 1]
```diff
@@ -879,7 +879,7 @@ func (s *PowerVSClusterScope) isDHCPServerActive() (bool, error) {
 		return false, err
 	}
 
-	requeue, err := s.checkDHCPServerStatus(dhcpServer.Status)
+	requeue, err := s.checkDHCPServerStatus(*dhcpServer)
```
Comment (Karthik-K-N) [inline cloud/scope/powervs_cluster.go:882]: How about logging the name here before calling the function? So we can avoid passing whole object.

File: cloud/scope/powervs_cluster.go [hunk 2]
```diff
@@ -2550,7 +2553,7 @@ func (s *PowerVSClusterScope) DeleteServiceInstance() (bool, error) {
 		return false, err
 	}
 	s.Info("PowerVS service instance successfully deleted")
-	return false, nil
+	return true, nil
```
Comment (Karthik-K-N) [inline cloud/scope/powervs_cluster.go:2556]: Why to re-queue in case of successful delete?

Comment (Amulyam24) [reply cloud/scope/powervs_cluster.go:2556]: As per the logic, we are requeuing after successful delete of a resource and wait for it to get deleted before checking again in the next reconcile to confirm if it has been removed successfully.

**Review by Karthik-K-N — COMMENTED** [review]
/lgtm

**Review by Prajyot-Parab — APPROVED** [review]

**Pull request merged**